tasks.py

```diff
@@ -0,0 +1,185 @@
+                'ContentModification/ExtendedGraphicStates/',
+                'ContentModification/FlattenTransparency/',
+                'ContentModification/LaunchActions/',
+                'ContentModification/MergePDF/',
+                'ContentModification/PageLabels/',
+                'ContentModification/PDFObject/',
+                'ContentModification/UnderlinesAndHighlights/',
+                'ContentModification/Watermark/',
+                'DocumentConversion/ColorConvertDocument/',
+                'DocumentConversion/ConvertToOffice/',
+                'DocumentConversion/CreateDocFromXPS/',
+                'DocumentConversion/FacturXConverter/',
+                'DocumentConversion/PDFAConverter/',
+                'DocumentConversion/PDFXConverter/',
+                'DocumentConversion/ZUGFeRDConverter/',
+                'DocumentOptimization/PDFOptimize/',
+                'Forms/ConvertXFAToAcroForms/',
+                'Forms/ExportFormsData/',
+                'Forms/FlattenForms/',
+                'Forms/ImportFormsData/',
+                'Images/DocToImages/',
+                'Images/DrawSeparations/',
+                'Images/EPSSeparations/',
+                'Images/GetSeparatedImages/',
+                'Images/ImageDisplayByteArray',
+                'Images/ImageEmbedICCProfile/',
+                'Images/ImageExport/',
+                'Images/ImageExtraction/',
+                'Images/ImageFromBufferedImage/',
+                'Images/ImageImport/',
+                'Images/ImageResampling/',
+                'Images/OutputPreview/',
+                'Images/RasterizePage/',
+                'InformationExtraction/ListBookmarks/',
+                'InformationExtraction/ListFonts/',
+                'InformationExtraction/ListInfo/',
+                'InformationExtraction/ListLayers/',
+                'InformationExtraction/ListPaths/',
+                'InformationExtraction/Metadata/',
+                'OpticalCharacterRecognition/AddTextToDocument/',
+                'OpticalCharacterRecognition/AddTextToImage/',
+                'Other/MemoryFileSystem/',
+                'Other/StreamIO/',
+                'Security/AddRegexRedaction/',
+                'Security/Redactions/',
+                'Text/AddGlyphs/',
+                'Text/AddUnicodeText/',
+                'Text/AddVerticalText/',
+                'Text/ListWords/',
+                'Text/RegexExtractText/',
+                'Text/RegexTextSearch/',
+                'Text/TextExtract/'
+        ]
+
+@task()
+def clean_samples(ctx):
+    """Cleans files that were generated from building the samples"""
+    for sample in samples_list:
+        full_path = os.path.join(os.getcwd(), sample)
+        with ctx.cd(full_path):
+            ctx.run('mvn clean')
+            ctx.run('git clean -fdx')
+
+@task()
+def build_samples(ctx):
+    """Builds the APDFL Java Maven samples"""
+    ctx.run('invoke clean-samples')
```

Comment:
Two things:

1. Since in the `Jenkinsfile` we're already running the `clean-samples` step before `build-samples`, I guess we want to make sure the repo is REALLY clean? 😄 This is probably okay, I'm just noticing the redundancy.
2. Instead of directly calling `ctx.run()`, Invoke has a capacity to declare prerequisites: a way to say, "always run _that_ task before _this_ task." You can look the feature up in the Invoke documentation, and see an example of it being used here: https://octocat.dlogics.com/datalogics/dl-conan-build-tools/blob/develop/src/dl_conan_build_tools/tasks/conan.py#L342

Reply:
1. Yes, we want the repo really clean because the files produced by building and running the samples take up a lot of resources.
2. Good idea, added!